speed-dreams/source-2.2.3/src/modules/userinterface/legacymenu/mainscreens/ResearcherMenu.cpp

```diff
@@ -327,6 +377,128 @@ static void OnActivate(void* /* dummy */)
     LoadDefaultSettings();
 }
 
+
+/// @brief Releases a file dialog
+void ReleaseDialogAndCom(IFileDialog* fileDialog)
+{
+    fileDialog->Release();
+    CoUninitialize();
+}
+
+/// @brief Releases a shell item and a file dialog
+void ReleaseShellItemAndDialogAndCom(IShellItem* shellItem, IFileDialog* fileDialog)
+{
+    shellItem->Release();
+    ReleaseDialogAndCom(fileDialog);
+}
+
+/// @brief Edits the black box button with a message, releases a shell item and file dialog
+void ErrorBlackBoxButtonThenReleaseShellItemAndDialogAndCom(const char* message, IShellItem* shellItem, IFileDialog* fileDialog)
+{
+    m_blackBoxChosen = false;
+    GfuiButtonSetText(s_scrHandle, m_blackBoxButton, message);
+    ReleaseShellItemAndDialogAndCom(shellItem, fileDialog);
+}
+
+/// @brief Opens a file dialog and changes the button to reflect this choice.
+static void SelectFile(void* /* dummy */)
+{
+    // Opens a file dialog on Windows
+
+    // Initialize COM interface
+    HRESULT hresult = CoInitializeEx(NULL, COINIT_APARTMENTTHREADED | COINIT_DISABLE_OLE1DDE);
+    if (FAILED(hresult)) { return; }
+
+    // Create a file dialog
+    IFileDialog* fileDialog;
+    hresult = CoCreateInstance(CLSID_FileOpenDialog, NULL, CLSCTX_ALL, IID_IFileDialog, reinterpret_cast<void**>(&fileDialog));
+    if (FAILED(hresult)) 
+    {
+    	CoUninitialize();
+    	return;
+    }
+
+    // Set file types for file dialog, then open it
+    COMDLG_FILTERSPEC filter[1] = { {L"Executables", L"*.exe"} };
+    hresult = fileDialog->SetFileTypes(1, filter);
+    if (FAILED(hresult))
+    {
+        ReleaseDialogAndCom(fileDialog);
+    	return;
+    }
+    hresult = fileDialog->Show(NULL);
+    if (FAILED(hresult))
+    {
+        ReleaseDialogAndCom(fileDialog);
+    	return;
+    }
+
+    // Get filename
+    IShellItem* shellItem;
+    hresult = fileDialog->GetResult(&shellItem);
+    if (FAILED(hresult))
+    {
+        ReleaseDialogAndCom(fileDialog);
+    	return;
+    } // I feel like I should release shellItem here as well, but the Microsoft Docs do not do so at this stage
+    PWSTR filePath;
+    hresult = shellItem->GetDisplayName(SIGDN_FILESYSPATH, &filePath);
+    if (FAILED(hresult))
+    {
+        ReleaseShellItemAndDialogAndCom(shellItem, fileDialog);
+    	return;
+    }
+
+    // Convert PWSTR to std::string
+    std::wstring_convert<std::codecvt_utf8<wchar_t>, wchar_t> converter;
+    std::string fileName = converter.to_bytes(filePath);
+    // Over max file length
+    if (fileName.size() >= BLACKBOX_PATH_SIZE - 1) // std::string isn't null terminated, while Windows paths/char* are
+    {
+        // Sanity check: This should be dead code: either your system is so old it does not support paths > 260 chars, or it has a system where paths of those lengths get aliased to an 8.3 file name that is <= 260 chars
```

Comment:
Comment probably runs off any normal sized screen, so continue on a new line
```suggestion
        // Sanity check: This should be dead code: either your system is so old it does not support paths > 260 chars, 
        // or it has a system where paths of those lengths get aliased to an 8.3 file name that is <= 260 chars
```